front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/AddPathStepPopup.tsx

```diff
@@ -27,44 +32,101 @@ type AddPathStepPopupProps = {
   resetFeatureInfoClick: () => void;
 };
 
-function AddPathStepPopup({
+const AddPathStepPopup = ({
   pathProperties,
   featureInfoClick,
   resetFeatureInfoClick,
-}: AddPathStepPopupProps) {
+}: AddPathStepPopupProps) => {
   const { getInfraID, getOrigin, getDestination } = useOsrdConfSelectors();
   const { launchPathfinding } = useManageTrainScheduleContext();
   const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
   const infraId = useSelector(getInfraID);
   const origin = useSelector(getOrigin);
   const destination = useSelector(getDestination);
 
+  const isOperationalPoint = useMemo(() => {
+    const properties = featureInfoClick?.feature?.properties;
+    return !!properties?.track_id || !!properties?.track_name;
+  }, [featureInfoClick]);
+
+  const { getTrackSectionsByIds } = useScenarioContext();
+
   const [trackOffset, setTrackOffset] = useState(0);
 
+  const [clickedOp, setClickedOp] = useState<
+    Extract<PathStep, { uic: number }> & {
+      tracks: {
+        trackName?: string;
+        coordinates?: number[];
+        isSelected: boolean;
+      }[];
+    }
+  >();
+
   const [getTrackEntity] =
     osrdEditoastApi.endpoints.postInfraByInfraIdObjectsAndObjectType.useLazyQuery();
 
   useEffect(() => {
     const calculateOffset = async () => {
-      const trackId = featureInfoClick.feature.properties?.id;
+      const objectId = featureInfoClick.feature.properties?.id;
+
       const result = await getTrackEntity({
         infraId: infraId!,
-        objectType: 'TrackSection',
-        body: [trackId],
+        objectType: isOperationalPoint ? 'OperationalPoint' : 'TrackSection',
+        body: [objectId],
       }).unwrap();
 
       if (!result.length) {
         console.error('No track found');
         return;
       }
 
-      const trackEntity = editoastToEditorEntity<TrackSectionEntity>(result[0], 'TrackSection');
-      const offset = calculateDistanceAlongTrack(
-        trackEntity,
-        point(featureInfoClick.coordinates.slice(0, 2)).geometry,
-        'millimeters'
-      );
-      setTrackOffset(offset);
+      if (isOperationalPoint) {
+        const trackId = featureInfoClick.feature.properties?.track_id;
+
+        const clickedTrack = await getTrackEntity({
+          infraId: infraId!,
+          objectType: 'TrackSection',
+          body: [trackId],
+        }).unwrap();
+
+        const { parts } = result[0].railjson as OperationalPoint;
```

Comment:
You can reuse this const l115 and 116 (we should avoid `any`s ;) )
```suggestion
        const operationalPoint = result[0].railjson as OperationalPoint;
```

Reply:
You're right, it's done